src/demuxer.rs

```diff
@@ -87,49 +83,27 @@ impl Demuxer for Y4MDemuxer {
     }
 }
 
-fn header_token(input: &[u8]) -> IResult<&[u8], &[u8]> {
-    take_till(|c| c == b' ' || c == b'\n')(input)
+fn from_utf8(input: &[u8]) -> std::result::Result<&str, std::str::Utf8Error> {
+    std::str::from_utf8(input)
 }
 
-fn header(input: &[u8]) -> IResult<&[u8], Y4MHeader> {
-    let (mut i, _) = tag("YUV4MPEG2 ")(input)?;
-    let mut width: usize = 0;
-    let mut height: usize = 0;
+fn header_token(input: &[u8]) -> IResult<&[u8], &str> {
+    map_res(take_till(|c| c == b' '), from_utf8)(input).map(|(i, token_str)| {
+        let (_, val) = token_str.split_at(1);
+        // Remove space from input
+        (&i[1..], val)
+    })
+}
 
+fn header(input: &[u8]) -> IResult<&[u8], Y4MHeader> {
+    let (i, _) = tag("YUV4MPEG2 ")(input)?;
 
-    loop {
-        let (ii, token) = header_token(i)?;
-        let token_str = std::str::from_utf8(&token).unwrap();
-        let (id, val) = token_str.split_at(1);
+    let (_i, (width, height, _framerate)) = tuple((header_token, header_token, header_token))(i)?;
 
-        match id.chars().next().unwrap() {
-            'W' => {
-                if let Ok(w) = val.parse::<usize>() {
-                    width = w;
-                }
-            }
-            'H' => {
-                if let Ok(h) = val.parse::<usize>() {
-                    height = h;
-                }
-            }
-            _ => {}
-        }
-
-        if ii[0] == b'\n'
-        {
-            break;
-        }
-        i = &ii[1..];
-    }
+    let width = width.parse::<usize>().unwrap_or(0);
```

Comment:
This does include the 'W' and probably won't work?

Reply:
We remove the "W" char here: 
```
let (_, val) = token_str.split_at(1);
// Remove space from input
(&i[1..], val)
})
```
so that is a number